core/server/data/migrations/versions/4.0/21-solve-orphaned-webhooks.js

```diff
@@ -0,0 +1,92 @@
+const {createIrreversibleMigration} = require('../../utils');
+const logging = require('../../../../../shared/logging');
+const ObjectID = require('bson-objectid');
+const crypto = require('crypto');
+
+// createSecret is copied from core/server/models/api-key.js
+const createSecret = (type) => {
+    const bytes = type === 'content' ? 13 : 32;
+    return crypto.randomBytes(bytes).toString('hex');
+};
+
+module.exports = createIrreversibleMigration(
+    async function up(knex) {
+        logging.info('Resolving the orphaned webhooks');
+
+        const orphanedWebhooks = await knex('webhooks')
+            .select('webhooks.id')
+            .leftJoin('integrations', 'integrations.id', 'webhooks.integration_id')
+            .where('integrations.id', null);
+
+        if (orphanedWebhooks.length === 0) {
+            logging.info('No orphaned webhooks found, skipping');
+            return;
+        }
+
+        const [role] = await knex('roles')
+            .select('roles.id')
+            .where('name', 'Admin Integration');
+
+        if (!role) {
+            logging.info('Failed to apply migration because of missing role.');
+            return;
+        }
+
+        const now = knex.raw('CURRENT_TIMESTAMP');
+        const id = ObjectID.generate();
+
+        const integration = {
+            id: id,
+            type: 'custom',
+            name: `Legacy webhooks`,
+            slug: `legacy-webhooks-${id}`,
+            icon_image: null,
+            description: `This integration was created as part of the 4.0 migration. It contains all webhooks created via the API that weren't visible in the admin interface previously.`,
+            created_at: now,
+            created_by: 1,
+            updated_at: now,
+            updated_by: 1
+        };
+
+        await knex('integrations')
+            .insert(integration);
+
+        const contentKey = {
```

Comment:
Out of interest, what's the reason for creating API keys for this integration?

Reply:
I haven't found a place in the code that created the API key, so I assumed that I had to do it by hand, given that I'm not using any code from the integration service.